### PR TITLE
Add Grid-Aligned behavior to Playlab

### PR DESCRIPTION
**Pull request by Hamms**
For our BB-8 levels; this is primarily to fix the behavior where BB-8 turns to face the null direction (south) between each movement step:

forward

Behavior that is _particularly_ funky when the solution involves using relative movement:

turning

To fix this, I added dedicated methods for turning BB-8 (and animating the turn) so that we can highlight and animate the block, and added a new behavior which we can apply when using these blocks so that BB-8 can avoid being faced south and being compelled to move, resulting in an animation that looks like this:

turningafter

### CODE REVIEW

File: apps/src/studio/studio.js [hunk 1]
```diff
@@ -5916,6 +5974,7 @@ Studio.moveSingle = function (opts) {
   }
 
   if (skin.gridAlignedMovement) {
+    sprite.setActivity(constants.BEHAVIOR_GRID_ALIGNED);
```
Comment (Hamms) [inline apps/src/studio/studio.js:5976]: I'd love to solicit some opinions on whether or not this is the correct place to set this behavior

File: apps/src/studio/studio.js [hunk 2]
```diff
@@ -5858,6 +5868,54 @@ Studio.getSkin = function () {
...
+Studio.turnSingle = function (opts) {
+  const sprite = Studio.sprite[opts.spriteIndex];
+  sprite.lastMove = Studio.tickCount;
+
+  if (skin.gridAlignedMovement) {
+    sprite.setActivity(constants.BEHAVIOR_GRID_ALIGNED);
+    sprite.addAction(new GridTurn(opts.dir, skin.slowExecutionFactor));
+    Studio.yieldGridAlignedTicks();
+  } else {
+    throw new TypeError("Studio.turnSingle is only valid in grid-aligned mode");
```
Comment (islemaster) [inline apps/src/studio/studio.js]: Style nit: You could check this condition and throw as an early-out at the top of this method, which would make it more obvious that the rest of the method is a simple command list with no branches.

```js
Studio.turnSingle = function (opts) {
  if (!skin.gridAlignedMovement) {
    throw new TypeError("Studio.turnSingle is only valid in grid-aligned mode");
  }

  const sprite = Studio.sprite[opts.spriteIndex];
  sprite.lastMove = Studio.tickCount;
  sprite.setActivity(constants.BEHAVIOR_GRID_ALIGNED);
  sprite.addAction(new GridTurn(opts.dir, skin.slowExecutionFactor));
  Studio.yieldGridAlignedTicks();
  Studio.lastMoveSingleDir = opts.dir;
};
```

File: apps/src/studio/studio.js (hunk 1 above)
Comment (islemaster) [reply apps/src/studio/studio.js:5976]: This seems good to me.  It's actually weirder that I don't see a corresponding `setActivity` call in the else case.

File: apps/test/unit/studio/itemTest.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+describe('item', () => {
+  before(() => {
+    Studio.trackedBehavior = {};
+    Studio.trackedBehavior.createdItems = {};
+  });
+
+  let item;
+
+  describe('update', () => {
+    describe('if destination is set', () => {
```
Comment (islemaster) [inline apps/test/unit/studio/itemTest.js:25]: Unnecessary nested `describe` blocks?

Comment (Hamms) [reply apps/test/unit/studio/itemTest.js:25]: sort of unnecessary, but it is the case that `update` can have some pretty different behavior based on when in the "lifecycle" of destination determination it's called, and I wanted to make absolutely clear which case we're in

File: apps/src/studio/studio.js (hunk 1 above)
Comment (Hamms) [reply apps/src/studio/studio.js:5976]: I _think_ it's the case that the else case expects `BEHAVIOR_STOP`, which is the default. If I'm correct then I could add that corresponding call for cleanliness, but if I'm not I have no idea what all could break

**Pull request merged**